operator/src/trustee.rs

```diff
@@ -426,3 +426,475 @@ pub async fn generate_kbs(
+
+        // 2. Create a mock client that returns 409 Conflict
+        let client = mock_client(StatusCode::CONFLICT, error_body.into_bytes());
+
+        // 3. Call the function under test
+        let result = create_reference_value_config_map(client, "test-ns", "test-rv-map").await;
+
+        // 4. Assert the result: Because the `info_if_exists!` macro catches the 409 error
+        // and treats it as non-fatal, we expect the function to still return Ok(()).
+        assert!(result.is_ok());
+    }
+
+    #[tokio::test]
+    async fn test_create_rv_config_map_generic_error() {
+        // 1. Prepare a response body for a 500 error
+        let error_response = ErrorResponse {
+            status: "Failure".to_string(),
+            message: "internal server error".to_string(),
+            reason: "ServerTimeout".to_string(),
+            code: 500,
+        };
+        let error_body = serde_json::to_string(&error_response).unwrap();
+
+        // 2. Create a mock client that returns 500 Internal Server Error
+        let client = mock_client(StatusCode::INTERNAL_SERVER_ERROR, error_body.into_bytes());
+
+        // 3. Call the function under test
+        let result = create_reference_value_config_map(client, "test-ns", "test-rv-map").await;
+
+        // 4. Assert the result: This time, we expect the function to return an error.
+        assert!(result.is_err());
+    }
+
+    // --- Tests for other simple creation functions ---
+    #[tokio::test]
+    async fn test_generate_resource_policy_success() {
+        let created_cm_json = serde_json::to_string(&ConfigMap {
+            metadata: ObjectMeta {
+                name: Some("test-policy".to_string()),
+                ..Default::default()
+            },
+            ..Default::default()
+        })
+        .unwrap();
+        let client = mock_client(StatusCode::OK, created_cm_json.into_bytes());
+
+        let result = generate_resource_policy(client, "test-ns", "test-policy").await;
+        assert!(result.is_ok());
+    }
+
+    #[tokio::test]
+    async fn test_generate_kbs_https_certificate_success() {
+        // This function creates two secrets. The mock client will be called twice.
+        // Since our simple mock is stateless, it will return the same success response both times.
+        let created_secret_json = serde_json::to_string(&Secret {
+            metadata: ObjectMeta {
+                name: Some("dummy-secret".to_string()),
+                ..Default::default()
+            },
+            ..Default::default()
+        })
+        .unwrap();
+        let client = mock_client(StatusCode::OK, created_secret_json.into_bytes());
+
+        let result = generate_kbs_https_certificate(client, "test-ns").await;
+        assert!(result.is_ok());
+    }
+
+    #[tokio::test]
+    async fn test_generate_kbs_configurations_success() {
+        // This function creates three configmaps in a loop.
```

Comment:
That's a true statement, but it's not very clear how it affects the test code below.

Reply:
Describe it a little more, please check if that works?